### PR TITLE
Split out Vercel jobs to be able to remove retry

### PR DESCRIPTION
**Pull request by janpio**
The Vercel jobs are currently often failing because of API limits being hit.
This is made worse by our automated retries which do not wait between jobs.
Until https://github.com/prisma/ecosystem-tests/issues/2728 makes us wait properly between attempts, this PR disables at least the retries to not make the situation worse.

### CODE REVIEW

**Comment by janpio** [comment]
- The failing tests are unrelated and fixed via https://github.com/prisma/ecosystem-tests/pull/2731
- This PR will go into conflicted state on each merge to `dev`, so I suggest you review/approve and I merge `dev` once before merging.

File: .github/scripts/check-folders.js
Comment (janpio) [inline .github/scripts/check-folders.js:6]: Small cleanup in this file to make it easier to understand, improvements on the comments welcome.

File: .github/workflows/test.yaml
Comment (janpio) [inline .github/workflows/test.yaml:786]: This is the important difference here that will leads to making the Vercel API situation less bad.

File: .github/scripts/check-folders.js
Comment (Jolg42) [inline .github/scripts/check-folders.js]: It's transforming the shape to return something that can be easy to use from GitHub Actions if conditionals

File: .github/scripts/check-folders.js
Comment (Jolg42) [inline .github/scripts/check-folders.js]: Create an array of job names, from reading the `jobs` keys of the .yml file

Example of an item should look like `platforms-serverless`, which is also matching the directory name

File: .github/scripts/check-folders.js
Comment (Jolg42) [reply .github/scripts/check-folders.js]: Looks like 
```
Object {
  "binaries": true,
  "bundlers": true,
  "core-features": true,
  "databases": true,
  "databases-macos": true,
  "dataproxy": true,
  "docker": true,
  "engines": true,
  "frameworks": true,
  "libraries": true,
  "migrate": true,
  "node": true,
  "os": true,
  "packagers": true,
  "platforms": true,
  "platforms-serverless": true,
  "platforms-serverless-vercel": true,
  "process-managers": true,
  "test-runners": true,
}
```
See https://github.com/prisma/ecosystem-tests/blob/dev/.github/workflows/detect-jobs-to-run.test.js

**Review by Jolg42 — APPROVED** [review]
That's a very nice improvement!

**Comment by Jolg42** [comment]
@janpio Looks like you will need to fix the conflicts and merge right away

**Comment by janpio** [comment]
Will do after applying your suggestions on the comments.

File: .github/scripts/check-folders.js
Comment (janpio) [reply .github/scripts/check-folders.js]:
```suggestion
  // Transform the shape to return something that can be easy to use from GitHub Actions if conditionals
```

File: .github/scripts/check-folders.js
Comment (janpio) [reply .github/scripts/check-folders.js]:
```suggestion
  // Create an array of job names, from reading the `jobs` keys of the .yml file
```

**Pull request merged**
